### PR TITLE
Feature/job expiration

### PR DESCRIPTION
**Pull request by gemantzu**

### CODE REVIEW

File: lib/mix/tasks/expire_jobs.ex [hunk 1]
```diff
@@ -0,0 +1,18 @@
+defmodule Mix.Tasks.ExpireJobs do
+  use Mix.Task
+  alias Companies.{Repo, Schema.Job}
+  import Ecto.Query, warn: false
+
+  @shortdoc "Expires all jobs that haven't been updated for a month."
+  def run(_) do
+    [:postgrex, :ecto]
+    |> Enum.each(&Application.ensure_all_started/1)
```
Comment (doomspork) [inline lib/mix/tasks/expire_jobs.ex:11]: No pipe here.

You could also just make this `Application.ensure_all_started(:companies)` and remove the start_link below.

File: lib/mix/tasks/expire_jobs.ex [hunk 2]
```diff
@@ -0,0 +1,18 @@
+defmodule Mix.Tasks.ExpireJobs do
+  use Mix.Task
+  alias Companies.{Repo, Schema.Job}
```
Comment (doomspork) [inline lib/mix/tasks/expire_jobs.ex:5]: Alias after import please. Separated by an empty line.

File: lib/companies/companies.ex [hunk 3]
```diff
@@ -20,12 +20,14 @@ defmodule Companies.Companies do
...
 
+    job_query = from j in Job, where: [expired: false]
+
     (c in Company)
     |> from()
     |> predicates(params)
     |> order_by(^order)
     |> where([c, _i, _j], is_nil(c.removed_pending_change_id))
-    |> preload([:industry, :jobs])
+    |> preload([:industry, jobs: ^job_query])
```
Comment (doomspork) [inline lib/companies/companies.ex:30]: Can we avoid this subquery by just adding a where clause to this?

```elixir
where([_c, _i, j], not j.expired)
```

**Comment by gemantzu** [comment]
Bump :)

**Comment by doomspork** [comment]
@gemantzu you need to rebase this branch.

**Pull request merged**